Scripts/Editor/UI/EditorInputDialog.cs

```diff
@@ -0,0 +1,178 @@
+            m_InputText = defaultInputText ?? string.Empty;
+        }
+
+        private void OnGUI()
+        {
+            Event currentEvent = Event.current;
+            CheckKeyboard(currentEvent);
+
+            Rect rect = EditorGUILayout.BeginVertical();
+
+            EditorGUILayout.Space(12);
+            EditorGUILayout.LabelField(m_Description);
+
+            EditorGUILayout.Space(8);
+            GUI.SetNextControlName(CONTROL_NAME_INPUT_TEXT);
+            m_InputText = EditorGUILayout.TextField(m_InputTextLabel, m_InputText);
+            GUI.FocusControl(CONTROL_NAME_INPUT_TEXT);
+            EditorGUILayout.Space(12);
+
+            Rect controlRect = EditorGUILayout.GetControlRect();
+            controlRect.width *= 0.5f;
+            if (GUI.Button(controlRect, m_OKBtnText))
+            {
+                Complete(false);
+            }
+
+            controlRect.x += controlRect.width;
+            if (GUI.Button(controlRect, m_CancelBtnText))
+            {
+                Complete(true);
+            }
+
+            EditorGUILayout.Space(8);
+            EditorGUILayout.EndVertical();
+
+            if (rect.width != 0 && minSize != rect.size)
+            {
+                minSize = maxSize = rect.size;
+            }
+
+            InitializePosition(currentEvent);
+        }
+
+        private void CheckKeyboard(Event currentEvent)
+        {
+            if (currentEvent.type != EventType.KeyDown)
+            {
+                return;
+            }
+            
+            // ReSharper disable once SwitchStatementMissingSomeEnumCasesNoDefault
+            switch (currentEvent.keyCode)
+            {
+                case KeyCode.Escape:
+                    currentEvent.Use();
+                    Complete(true);
+                    break;
+
+                case KeyCode.Return:
+                case KeyCode.KeypadEnter:
+                    currentEvent.Use();
+                    Complete(false);
+                    break;
+            }
+        }
+
+        private void InitializePosition(Event currentEvent)
+        {
+            if (m_HasInitializedPosition || currentEvent.type != EventType.Layout)
+            {
+                return;
+            }
+            m_HasInitializedPosition = true;
+
+            Vector2 mousePos = GUIUtility.GUIToScreenPoint(currentEvent.mousePosition);
+            mousePos.x += 32;
+            if (mousePos.x + position.width > m_MaxScreenPosition.x)
+            {
+                mousePos.x -= position.width + 64;
+            }
+            if (mousePos.y + position.height > m_MaxScreenPosition.y)
+            {
+                mousePos.y = m_MaxScreenPosition.y - position.height;
+            }
+
+            position = new Rect(mousePos.x, mousePos.y, position.width, position.height);
+
+            Focus();
+        }
+
+        private void Complete(bool isCancel)
+        {
+            OnComplete?.Invoke(isCancel ? string.Empty : m_InputText);
```

Comment:
Would it be better to communicate cancel and let the developer read the string off the instance instead?

Reply:
Yes, refactored